dqc/test/test_properties.py

```diff
@@ -441,3 +441,26 @@ def get_jac_ene(atomposs, efield, grad_efield):
         # raman spectra intensities
         torch.autograd.gradgradcheck(get_jac_ene, (atomposs.detach(), efield, grad_efield.detach()),
                                      atol=3e-4)
+
+def test_optimal_geometry(h2o_qc):
+    # test if the optimal geometry of h2o similar to pyscf
+    # from CCCBDB (calculated geometry for H2O)
+    pyscf_h2o_opt = h2o_qc.get_system().atompos
+
+    # create a new h2o with poor initial geometry
+    h2o_init = torch.tensor([
+        [0.0, 0.0, 0.215],
+        [0.0, 1.475, -0.863],
+        [0.0, -1.475, -0.863],
+    ], dtype=dtype).requires_grad_()
+   
+    # check starting geometry is not optimal
+    assert not torch.allclose(h2o_init, pyscf_h2o_opt, rtol=2e-4)
+    
+    # optimize geometry
+    system = h2o_qc.get_system()
+    new_system = system.make_copy(moldesc=(system.atomzs, system.atompos))
+    new_qc = h2o_qc.__class__(new_system).run()
+    h2o_opt = optimal_geometry(new_qc)
+
+    assert torch.allclose(h2o_opt, pyscf_h2o_opt, rtol=2e-4)
```

Comment:
Does this always guaranteed? There are infinitely many optimal geometry for H2O (i.e. because of translational and rotational symmetry), so if we change the optimization procedure in the future, this checking might break down because of those symmetries.
Maybe checking the bond lengths would make it more robust?

Reply:
I have changed this to check bond lengths

Reply:
Thanks